### ISSUE TITLE
Compiler hanging

### ISSUE THREAD

**Issue by gergelyk** [rust-lang/rust#78016]
I tried this code:

```rust
use std::io::{self, Read};

fn main() {
    let mut buffer = String::new();
    io::stdin().read_to_string(&mut buffer).unwrap();
}
```

After running `cargo run`...

I expected to see this happen: *executable gets compiled or compiler exits with an error*

Instead, this happened: *compiler keeps hanging for several seconds*

`rustc --version --verbose`:
```
rustc 1.47.0
binary: rustc
commit-hash: unknown
commit-date: unknown
host: x86_64-unknown-linux-gnu
release: 1.47.0
LLVM version: 10.0

```

**Comment by gergelyk** [comment]
False alarm. That was application waiting for stdin :) Let's forget about this "issue".
